crates/core/src/models/felt.rs

```diff
@@ -38,17 +43,26 @@ impl From<u64> for Felt252Wrapper {
     }
 }
 
-impl From<H160> for Felt252Wrapper {
-    fn from(h160: H160) -> Self {
-        let felt = FieldElement::from_byte_slice_be(&h160.0).unwrap(); // safe unwrap since H160 is 20 bytes
+impl From<Address> for Felt252Wrapper {
+    fn from(address: Address) -> Self {
+        let felt = FieldElement::from_byte_slice_be(&address.0).unwrap(); // safe unwrap since H160 is 20 bytes
         Self(felt)
     }
 }
 
-impl From<Felt252Wrapper> for H160 {
-    fn from(felt: Felt252Wrapper) -> Self {
+impl TryFrom<Felt252Wrapper> for Address {
+    type Error = Felt252WrapperError;
```

Comment:
we want conversion to fail when we are going from felt252wrapper to address assuming

- the felt comes from our kkrt evm in the places where it is expected to properly go from hash -> address
  - https://github.com/kkrt-labs/kakarot/blob/1a52997544ed0f00751e5ba10970d7a8c2c51b32/src/utils/utils.cairo#L792-L799

so if we are getting a felt that is larger than the expected amount, we should throw an error to signal that something is off sooner rather than later

Reply:
yes